src/geometry_integration.jl

```diff
@@ -23,17 +24,17 @@ function RowCols(ncols::Int, nrows::Int)
     )
 end
 
-Base.getindex(rowcols::RowCols, ::Left) = rowcols.lefts
-Base.getindex(rowcols::RowCols, ::Right) = rowcols.rights
-Base.getindex(rowcols::RowCols, ::Top) = rowcols.tops
-Base.getindex(rowcols::RowCols, ::Bottom) = rowcols.bottoms
+Base.getindex(rowcols::RowCols, side::Side) = side == Left ? rowcols.lefts :
+                                              side == Right ? rowcols.rights :
+                                              side == Top ? rowcols.tops :
+                                              side == Bottom ? rowcols.bottoms : throw_side(side)

```

Comment:
why is this style better than the dispatch version? aren't four methods compiled anyway?

Reply:
ah I see now that you've changed it to an enum

Reply:
so this style with a couple of types should be avoided in general?

Reply:
It depends: if `Side` is effectively "closed" then an enum has a big advantage, but a big weakness if your goal is external extensibility. More detail: recall that Julia specializes methods for each specific argument type:
```julia
julia> abstract type StaticNumber <: Number end

julia> struct One <: StaticNumber end

julia> struct Two <: StaticNumber end

julia> f(n::StaticNumber) = nothing
f (generic function with 1 method)

julia> f(One())

julia> f(Two())

julia> using MethodAnalysis

julia> methodinstances(f)
2-element Vector{Core.MethodInstance}:
 MethodInstance for f(::One)
 MethodInstance for f(::Two)
```
That means `f` has to be compiled twice. But with this version:
```julia
julia> @enum StaticNumber One Two

julia> f(n::StaticNumber) = nothing
f (generic function with 1 method)

julia> f(One)

julia> f(Two)

julia> using MethodAnalysis

julia> methodinstances(f)
1-element Vector{Core.MethodInstance}:
 MethodInstance for f(::StaticNumber)
```
it only has to be compiled once. If you have several such arguments to a multi-argument function, the resulting savings/costs can be *quite* large. This will presumably be more impactful in Makie itself which adds a lot more types.

The downside is that you can no longer use dispatch to extend `Side` unless you do it in this package. As long as you're OK with that, the advantages here are surely worth having.